NuRadioReco/modules/channelSineSubtraction.py

```diff
@@ -0,0 +1,249 @@
+
+import numpy as np
+import sys
+import logging
+# Add the directory where your library is located
+#sys.path.append('/Users/nozdrina.1/Documents/phys/software/NuRadioMC')
+
+import NuRadioReco.utilities as ut
+import matplotlib.pyplot as plt
+from scipy.optimize import curve_fit
+from NuRadioReco.utilities import units, fft
+
+"""
+Contains module to filter continuous wave out of the signal using sine subtraction on peaks in frequency spectrum
+"""
+
+
+def guess_amplitude(wf: np.ndarray, target_freq: float, sampling_rate: float = 3.2):
+    """
+        Estimate the amplitude of a specific harmonic in the waveform.
+
+        Args:
+            wf (np.ndarray): Input waveform (1D array).
+            target_freq (float): Target frequency for which to estimate amplitude.
+            sampling_rate (float): Sampling rate of the waveform (GHz).
+
+        Returns:
+            float: Estimated amplitude of the target frequency.
+        """
+    if wf.size == 0:
+        raise ValueError("Input waveform is empty.")
+    if target_freq < 0 or target_freq > sampling_rate / 2:
+        raise ValueError("Target frequency is out of range (0 to Nyquist frequency).")
+
+    fft_spectrum = np.fft.fft(wf)
+    frequencies = np.fft.fftfreq(len(wf), d=1 / sampling_rate)
+
```

Comment:
We should use the functions defined in `NuRadioReco.utilities.fft` as also done elsewhere in this module already. If we do that we also do not have to fix the amplitude below (i.e., the normalisation should be fine with no extra effort).